torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4653,40 +4654,70 @@ def sample_inputs_cholesky_solve(op_info, device, dtype, requires_grad=False, **
 
 
 def sample_inputs_lu(op_info, device, dtype, requires_grad=False, **kwargs):
+    make_arg = partial(make_fullrank_matrices_with_distinct_singular_values,
+                       dtype=dtype, device=device, requires_grad=requires_grad)
+
     # not needed once OpInfo tests support Iterables
     def generate_samples():
         batch_shapes = ((), (3,), (3, 3))
         for batch_shape, get_infos, size_delta in product(batch_shapes, (True, False), (-2, -1, 0, +1, +2)):
             shape = batch_shape + (S + size_delta, S)
-            input = make_tensor(shape, device, dtype, requires_grad=requires_grad, low=None, high=None)
+            input = make_arg(*shape)
             yield SampleInput(input, args=(True, get_infos))
 
     return list(generate_samples())
 
+def sample_inputs_linalg_lu_factor(op_info, device, dtype, requires_grad=False, **kwargs):
+    # When calling `lu_factor` we need to assure that the matrix is invertible
+    make_fn = make_tensor if "ex" in op_info.name else make_fullrank_matrices_with_distinct_singular_values
+    make_arg = partial(make_fn, dtype=dtype, device=device, requires_grad=requires_grad)
+
+    # not needed once OpInfo tests support Iterables
+    def generate_samples():
+        batch_shapes = ((), (3,), (3, 3))
+        # pivot=False only supported in CUDA
+        pivots = (True, False) if device == "cuda" else (True,)
```

Comment:
This isn't correct. While for the lu test in test_linalg it's OK to compare device with the string "cpu", for CUDA devices the string is going to be "cuda:0" or "cuda:1". The device needs to be turned into a torch device and then the device type needs to be checked:

```
torch.device(device).device_type == 'cuda'
```

Reply:
Yes I know this is a little awkward and it'd probably be nice to change the "device" arg passed to our tests to actually be a `torch.device`.